tests/test_web_tools.py

```
@@ -61,3 +72,173 @@ def respond(request: httpx.Request) -> httpx.Response:
+    def respond(request: httpx.Request) -> httpx.Response:
+        requested.append(request.url.path)
+        return httpx.Response(
+            302,
+            headers={"location": "//127.0.0.1/private"},
+            request=request,
+        )
+
+    monkeypatch.setattr(web_tools, "resolve_public_http_url", resolve)
+
+    with httpx.Client(transport=httpx.MockTransport(respond)) as client:
+        with pytest.raises(ValueError, match="non-public"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)